internal/background/availability_queue.go

```diff
@@ -59,8 +85,26 @@ func sendAvailabilityRequestMessages(ctx context.Context, batchSize int, tickDur
 				logger.Trace().Int("messages", length).Msgf("Sending %d availability request messages (cancel)", length)
 				send(ctx, messageBuffer...)
 			}
-
 			return
 		}
 	}
 }
+
+func IsProvisioningSource(ctx context.Context, info availabilityRequestInfo, sc clients.Sources) (*kafka.GenericMessage, error) {
+	logger := ctxval.Logger(ctx)
+	logger.Trace().Msgf("Checking if the provisioning app is enabled for the given source id %s", info.asm.SourceID)
+
+	msg, err := info.asm.GenericMessage(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert to generic message %w", err)
+	}
+
+	_, err = sc.GetAuthentication(ctx, info.asm.SourceID)
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			return nil, fmt.Errorf("could not get source authentication: %w", err)
+		}
+		return nil, fmt.Errorf("could not get source authentication: %w", err)
+	}
+	return &msg, nil
+}
```

Comment:
Why this function is named IsProvisioningSource while it also has a second responsibility of converting a message?

It is more clear to first convert the message at the caller site and then pass the required input (source id) as an argument. Then it becomes a single responsibility function that really just checks the source.